### PR TITLE
Fix error messages when objects or buckets are not found.

### PR DESCRIPTION
**Pull request by quoctruong**
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-powershell/376)

### CODE REVIEW

File: Google.PowerShell/Storage/GcsObject.cs [hunk 1]
```diff
@@ -832,10 +841,25 @@ protected override void ProcessRecord()
...
+                CheckForError(result);
+            }
+        }
+
+        private void CheckForError(IDownloadProgress downloadProgress)
+        {
+            if (downloadProgress.Status == DownloadStatus.Failed || downloadProgress.Exception != null)
+            {
+                string exceptionMessage = downloadProgress.Exception.Message;
+                if (exceptionMessage != null && exceptionMessage.IndexOf("Not Found", System.StringComparison.OrdinalIgnoreCase) >= 0)
```
Comment (iantalarico) [inline Google.PowerShell/Storage/GcsObject.cs]: Is this a safe check? Shouldn't we check and error code? 

exceptionMessage.IndexOf("Not Found"

**Comment by quoctruong** [comment]
@iantalarico PTAL

File: Google.PowerShell/Storage/GcsObject.cs [hunk 2]
```diff
@@ -849,8 +849,8 @@ private void CheckForError(IDownloadProgress downloadProgress)
         {
             if (downloadProgress.Status == DownloadStatus.Failed || downloadProgress.Exception != null)
             {
-                string exceptionMessage = downloadProgress.Exception.Message;
-                if (exceptionMessage != null && exceptionMessage.IndexOf("Not Found", System.StringComparison.OrdinalIgnoreCase) >= 0)
+                GoogleApiException googleApiException = downloadProgress.Exception as GoogleApiException;
```
Comment (iantalarico) [inline Google.PowerShell/Storage/GcsObject.cs]: Do you need to check if this is a GoogleApiException or will it always be?

Comment (quoctruong) [inline Google.PowerShell/Storage/GcsObject.cs]: I have a check in line 853 so if it is not a GoogleApiException, we will just throw whatever exception message that it has.

**Comment by quoctruong** [comment]
Thanks!

**Pull request merged**